pkg/api/v20170701/validate.go

```diff
@@ -111,9 +111,6 @@ func (a *AgentPoolProfile) Validate(orchestratorType string) error {
 func (l *LinuxProfile) Validate() error {
 	// Don't need to call validate.Struct(l)
 	// It is handled by Properties.Validate()
-	if e := validate.Var(l.SSH.PublicKeys[0].KeyData, "required"); e != nil {
```

Comment:
we shouldn't touch this file, this apiversion is for ACS legacy support. This won't affect generate/deploy with vlabs in aks-engine.

Reply:
Reverted the file.